x-pack/plugins/security_solution/server/endpoint/services/actions/clients/crowdstrike/crowdstrike_actions_client.ts

```diff
@@ -0,0 +1,323 @@
+      const connector = connectorList.find(({ actionTypeId, isDeprecated, isMissingSecrets }) => {
+        return actionTypeId === CROWDSTRIKE_CONNECTOR_ID && !isDeprecated && !isMissingSecrets;
+      });
+
+      if (!connector) {
+        throw new ResponseActionsClientError(
+          `No Crowdstrike stack connector found`,
+          400,
+          connectorList
+        );
+      }
+
+      this.log.debug(`Using Crowdstrike stack connector: ${connector.name} (${connector.id})`);
+
+      return connector;
+    });
+  }
+
+  protected async writeActionRequestToEndpointIndex(
+    actionRequest: Omit<ResponseActionsClientWriteActionRequestToEndpointIndexOptions, 'hosts'>
+  ): Promise<LogsEndpointAction> {
+    const agentId = actionRequest.endpoint_ids[0];
+    const agentDetails = await this.getAgentDetails(agentId);
+
+    return super.writeActionRequestToEndpointIndex({
+      ...actionRequest,
+      hosts: {
+        [agentId]: { name: agentDetails[0].hostname },
+      },
+    });
+  }
+
+  /**
+   * Sends actions to Crowdstrike directly (via Connector)
+   * @private
+   */
+  private async sendAction(
+    actionType: SUB_ACTION,
+    actionParams: object
+  ): Promise<ActionTypeExecutorResult<unknown>> {
+    const { id: connectorId } = await this.getConnector();
+    const executeOptions: Parameters<typeof this.connectorActionsClient.execute>[0] = {
+      actionId: connectorId,
+      params: {
+        subAction: actionType,
+        subActionParams: actionParams,
+      },
+    };
+
+    this.log.debug(
+      `calling connector actions 'execute()' for Crowdstrike with:\n${stringify(executeOptions)}`
+    );
+
+    const actionSendResponse = await this.connectorActionsClient.execute(executeOptions);
+
+    if (actionSendResponse.status === 'error') {
+      this.log.error(stringify(actionSendResponse));
+
+      // TODO TC: check if we want to handle different errors like 401
+      throw new ResponseActionsClientError(
+        `Attempt to send [${actionType}] to Crowdstrike failed: ${
+          actionSendResponse.serviceMessage || actionSendResponse.message
+        }`,
+        500,
+        actionSendResponse
+      );
+    }
+
+    this.log.debug(`Response:\n${stringify(actionSendResponse)}`);
+
+    return actionSendResponse;
+  }
+
+  private async getAgentDetails(
+    id: string
+  ): Promise<CrowdstrikeGetAgentsResponse['resources'][number]> {
+    const { id: connectorId } = await this.getConnector();
+    const executeOptions: CrowdstrikeConnectorExecuteOptions<CrowdstrikeGetAgentsParams> = {
+      actionId: connectorId,
+      params: {
+        subAction: SUB_ACTION.GET_AGENT_DETAILS,
+        subActionParams: {
+          ids: [id],
+        },
+      },
+    };
+
+    let crowdstrikeApiResponse: CrowdstrikeGetAgentsResponse | undefined;
+
+    try {
+      const response = (await this.connectorActionsClient.execute(
+        executeOptions
+      )) as ActionTypeExecutorResult<CrowdstrikeGetAgentsResponse>;
```

Comment:
just FYI: with my PR we now have proper types on the `execute()` method. I made it a generic and thus we can set the response type when calling it and no longer have to cast response.